Automattic/jetpack
# Writing prompt block: add spacing support

## Proposed changes:

* Adds padding and top/bottom margin support to writing prompts block.
* Updates [block api to v2](https://make.wordpress.org/core/2020/11/18/block-api-version-2/), which is the correct api to use with the `useBlockProps` hook. This remove the extra wrapper div around the block and injects top level classes and styles into the main block div.

<img width="286" alt="image" src="https://user-images.githubusercontent.com/1699996/225680150-6590b85e-03af-425a-ba70-4a303362b2a6.png">

### Other information:


## Jetpack product discussion

PT: pe7F0s-uo-p2

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

Note, the label and footer of the block will be blank unless applying API updates from jetpack trunk to wpcom.

- Sandbox public-api and apply jetpack changes to your sandbox: `bin/jetpack-downloader test jetpack trunk`
- Make sure your site is connected to WP.com and experimental blocks are enabled (`define( 'JETPACK_EXPERIMENTAL_BLOCKS', true );`)
- Insert a writing prompt block. Go to the style settings and see Padding and Margin are added.
- Change the settings, see that the block in the editor applys the settings correctly.
- Save the block and see that margin and padding settings display correctly on the front end.
